apps/web/components/AdditionalCalendarSelector.tsx

```diff
@@ -0,0 +1,80 @@
+import React from "react";
+import Select from "react-select";
+import { OptionProps } from "react-select";
+
+import { InstallAppButton } from "@calcom/app-store/components";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import type { App } from "@calcom/types/App";
+import { Button } from "@calcom/ui";
+
+import { QueryCell } from "@lib/QueryCell";
+import { trpc, inferQueryOutput } from "@lib/trpc";
+
+interface Props {
+  isLoading?: boolean;
+}
+
+const ImageOption = (props: OptionProps) => {
+  const data = props.data as { [key: string]: string; type: App["type"] };
```

Comment:
Try using the generic instead of casting first:

```suggestion
const ImageOption = (props: OptionProps<{ [key: string]: string; type: App["type"] }>) => {
  const { data } = props;
```